explorer/app/views/Project/project.tsx

```diff
@@ -1,30 +1,46 @@
 // Core dependencies
 import React from "react";
+import { useRouter } from "next/router";
 
 // App dependencies
 import { ComponentCreator } from "app/components/ComponentCreator/ComponentCreator";
 import { Project as ProjectView } from "app/components/Project/project";
 import { useCurrentEntity } from "app/hooks/useCurrentEntity";
 import { useFetchEntity } from "app/hooks/useFetchEntity";
 import { DetailModel } from "../../models/viewModels";
+import { useCurrentDetailTab } from "app/hooks/useCurrentDetailTab";
 
 export const Project = (props: DetailModel) => {
   const { response, isLoading } = useFetchEntity(props);
+  const { push, query } = useRouter();
+  const uuid = query.uuid?.[0];
   const entity = useCurrentEntity();
-  const mainColumn = entity?.detail?.mainColumn;
-  const sideColumn = entity?.detail?.sideColumn;
+  const { currentTab, tabIndex } = useCurrentDetailTab(entity);
+  const mainColumn = currentTab?.mainColumn;
+  const sideColumn = currentTab?.sideColumn;
   const top = entity?.detail?.top;
+  const tabs = entity?.detail?.tabs.map((tab) => ({ label: tab.label }));
 
   if (isLoading || !response) {
     return <span>LOADING...</span>; //TODO: return the loading UI component
   }
 
-  if (!mainColumn || !sideColumn || !top) {
+  if (!mainColumn || !sideColumn || !top || !tabs || !currentTab) {
```

Comment:
Could you refactor this into a function along the lines of `isProjectConfigured`?

Reply:
Hey @MillenniumFalconMechanic ! I'm not sure if this end up like you imagined, but I was able to get rid of most of these guards by making sure the entity is not nullable. I'm doing that by only checking on the page component and changing the `useCurrentEntity` hook to make use of a not nullable context. Let me know what you think